src/overlays/actors/ovl_Boss_Dodongo/z_boss_dodongo.c

```diff
@@ -51,311 +44,1721 @@ const ActorInit Boss_Dodongo_InitVars = {
+
+        mtxScaleZ = 1.0f;
+        mtxScaleY = 1.0f;
+
+        if ((limbIndex == 33) || (limbIndex == 48)) {
+            mtxScaleY = mtxScaleZ = this->unk_1F8;
+        }
+
+        Matrix_Push();
+        Matrix_Scale(1.0f, mtxScaleY, mtxScaleZ, MTXMODE_APPLY);
+
+        if ((limbIndex != 6) && (limbIndex != 7)) {
+            Matrix_RotateX(this->unk_25C[limbIndex] * 0.115f, MTXMODE_APPLY);
+            Matrix_RotateY(this->unk_25C[limbIndex] * 0.13f, MTXMODE_APPLY);
+            Matrix_RotateZ(this->unk_25C[limbIndex] * 0.1f, MTXMODE_APPLY);
+            Matrix_Scale(1.0f - this->unk_208, this->unk_208 + 1.0f, 1.0f - this->unk_208, MTXMODE_APPLY);
+            Matrix_RotateZ(-(this->unk_25C[limbIndex] * 0.1f), MTXMODE_APPLY);
+            Matrix_RotateY(-(this->unk_25C[limbIndex] * 0.13f), MTXMODE_APPLY);
+            Matrix_RotateX(-(this->unk_25C[limbIndex] * 0.115f), MTXMODE_APPLY);
+        }
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_dodongo.c", 3822),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, *dList);
+        Matrix_Pop();
+
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_dodongo.c", 3826);
+    }
+    { s32 pad; } // Required to match
+    return 1;
+}
+
+void BossDodongo_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    static Vec3f D_808CA450 = { 5000.0f, -2500.0f, 0.0f };
+    static Vec3f D_808CA45C = { 0.0f, 0.0f, 0.0f };
+    static Vec3f D_808CA468 = { 11500.0f, -3000.0f, 0.0f };
+    static Vec3f D_808CA474 = { 5000.0f, -2000.0f, 0.0f };
+    static Vec3f D_808CA480 = { 8000.0f, 0.0f, 0.0f };
+    static Vec3f D_808CA48C = { 8000.0f, 0.0f, 0.0f };
+    BossDodongo* this = THIS;
+
+    if (limbIndex == 6) {
+        Matrix_MultVec3f(&D_808CA45C, &this->vec);
+        Matrix_MultVec3f(&D_808CA450, &this->actor.focus.pos);
+        Matrix_MultVec3f(&D_808CA468, &this->firePos);
+        Matrix_MultVec3f(&D_808CA474, &this->mouthPos);
+    } else if (limbIndex == 39) {
+        Matrix_MultVec3f(&D_808CA480, &this->unk_410);
+    } else if (limbIndex == 46) {
+        Matrix_MultVec3f(&D_808CA48C, &this->unk_404);
+    }
+    Collider_UpdateSpheres(limbIndex, &this->collider);
+}
+
+void BossDodongo_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BossDodongo* this = THIS;
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_dodongo.c", 3922);
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    if ((this->unk_1C0 >= 2) && ((this->unk_1C0 & 1) != 0)) {
+        POLY_OPA_DISP = Gfx_SetFog(POLY_OPA_DISP, 255, 255, 255, 0, 900, 1099);
+    } else {
+        POLY_OPA_DISP =
+            Gfx_SetFog(POLY_OPA_DISP, (u32)this->fogR, (u32)this->fogG, (u32)this->fogB, 0, this->fogMin, this->fogMax);
+    }
+
+    Matrix_RotateZ(this->unk_23C, 1);
+    Matrix_RotateX((this->unk_1C4 / 32768.0f) * 3.14159f, MTXMODE_APPLY);
```

Comment:
Does this work?
```suggestion
    Matrix_RotateX(BINANG_TO_RAD(this->unk_1C4), MTXMODE_APPLY);
```

Reply:
Unless this macro is incorrect it needs to be in the original form to match